packages/flutter/lib/src/material/segmented_button.dart

```diff
@@ -0,0 +1,770 @@
+  ///
+  /// The callback's parameter indicates which of the segments are selected.
+  ///
+  /// When the callback is null, the entire segmented button is disabled,
+  /// and will not respond to input.
+  final void Function(Set<T>)? onSelectionChanged;
+
+  /// Determines if multiple segments can be selected at one time.
+  ///
+  /// If true, more than one segment can be selected. When selecting a
+  /// segment, the other selected segments will stay selected. Selecting an
+  /// already selected segment will unselect it.
+  ///
+  /// If false, only one segment will be selected at a time. When a segment
+  /// is selected, any previously selected segment will be unselected.
+  final bool multiSelectionEnabled;
+
+  /// Determines if having no selected segments is allowed.
+  final bool emptySelectionAllowed;
+
+  /// Customizes this button's appearance.
+  ///
+  /// The following style properties apply to the entire segmented button:
+  ///
+  ///   * [ButtonStyle.shadowColor]
+  ///   * [ButtonStyle.elevation]
+  ///   * [ButtonStyle.side] - which is used for both the outer shape and
+  ///     dividers between segments.
+  ///   * [ButtonStyle.shape]
+  ///
+  /// The following style properties are applied to each of the invidual
+  /// button segments. For properties that are a [MaterialStateProperty],
+  /// they will be resolved with the current state of the segment:
+  ///
+  ///   * [ButtonStyle.textStyle]
+  ///   * [ButtonStyle.backgroundColor]
+  ///   * [ButtonStyle.foregroundColor]
+  ///   * [ButtonStyle.overlayColor]
+  ///   * [ButtonStyle.surfaceTintColor]
+  ///   * [ButtonStyle.elevation]
+  ///   * [ButtonStyle.padding]
+  ///   * [ButtonStyle.iconColor]
+  ///   * [ButtonStyle.iconSize]
+  ///   * [ButtonStyle.mouseCursor]
+  ///   * [ButtonStyle.visualDensity]
+  ///   * [ButtonStyle.tapTargetSize]
+  ///   * [ButtonStyle.animationDuration]
+  ///   * [ButtonStyle.enableFeedback]
+  ///   * [ButtonStyle.alignment]
+  ///   * [ButtonStyle.splashFactory]
+  final ButtonStyle? style;
+
+  /// Determines if the [selectedIcon] (usually a checkmark) is displayed on
+  /// the selected segments.
+  ///
+  /// If true the [selectedIcon] will be displayed at the start of the segment.
+  /// If both the [ButtonSegment.label] and [ButtonSegment.icon] are provided,
+  /// then the icon will be replaced with the [selectedIcon]. If only the icon
+  /// or the label is present then the [selectedIcon] will be shown at the start
+  /// of the segment.
+  ///
+  /// If false, then the [selectedIcon] is not used and will not be displayed
+  /// on selected segments.
+  final bool showSelectedIcon;
+
+  /// An icon that is used to indicate a segment is selected.
+  ///
+  /// If [showSelectedIcon] is true then for selected segments this icon
+  /// will be shown before the [ButtonSegment.label], replacing the
+  /// [ButtonSegment.icon] if it is specified.
+  ///
+  /// Defaults to `const Icon(Icons.check)`.
```

Comment:
It would be nice if there was a link to `[Icons.check]` so they could see it in the IDE, or click to it in the docs to see what it looks like.